source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -67,8 +67,12 @@ void UdpStatsdSink::flush(Stats::MetricSnapshot& snapshot) {
 }
 
 void UdpStatsdSink::onHistogramComplete(const Stats::Histogram& histogram, uint64_t value) {
-  // For statsd histograms are all timers.
-  const std::string message(fmt::format("{}.{}:{}|ms{}", prefix_, getName(histogram),
+  // For statsd histograms are all timers, append the ASCII SI symbol to disambiguate.
+  std::string si_name = histogram.unitSymbol().length() > 0
```

Comment:
taste-test: WDYT of writing this as:
```
std::string si_name(getName(histogram));
if (histogram.unit() != Histogram::Unit::Undefined) {
  absl::StrAppend(&si_name, "_", histogramUnitSymbol();
}
```

This seems easier to read than the ternary, and avoids calling unitSymbol() twice, which (in current PR) returns a string, causing an alloca and a copy.

Reply:
Fixed, no strong opinion on the readability (slightly like ternaries better here) but definitely agree on the double invocation.